modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -106,6 +107,15 @@ export default class Ecdsa {
     // Add secret shares
     const x = allShares.map((participant) => hexToBigInt(participant['u'])).reduce(Ecdsa.curve.scalarAdd);
 
+    // Verify shares.
+    for (const share of nShares) {
+      try {
+        Ecdsa.shamir.verify(hexToBigInt(share.u), [hexToBigInt(share.y), hexToBigInt(share.v)], pShare.i);
+      } catch (err) {
+        throw new Error(`Could not verify share from participant ${share.j}`);
```

Comment:
NIT: Can we add actual the error thrown also to to the message so that it will also give info on what check at which the verification actually failed.
```suggestion
        throw new Error(`Could not verify share from participant ${share.j}. Verification Error: ${err}`);
```

Reply:
Done